sdk/core/azure-core/src/main/java/com/azure/core/http/policy/PortPolicy.java

```diff
@@ -40,7 +40,8 @@ public PortPolicy(int port, boolean overwrite) {
             try {
                 context.getHttpRequest().setUrl(urlBuilder.setPort(port).toURL());
             } catch (MalformedURLException e) {
-                return Mono.error(e);
+                return Mono.error(new RuntimeException(
+                    String.format("Failed to set port %d to http request.", port), e));
```

Comment:
Failed to set the HTTP request port to %d.

Thoughts on this? It feels a little more fluid in wording.

Reply:
Done.